### PR TITLE
Fix: jumpy composer

### LINKED ISSUE #15981: Composer jumpy while maximized
# Bug

The user can see the composer max height jumping up and down when removing an image while maximized. The same effect is visible with link previews, therefore the problem is not specific about one type of *extra* content on the composer.

[composer-bug-height.webm](https://github.com/status-im/status-mobile/assets/46027/915095c6-8bb7-49cd-a0ff-960f4ee4d8a2)

[composer-bug-height2.webm](https://github.com/status-im/status-mobile/assets/46027/7a1a2916-0570-4b94-8474-3013bba87097)

### Solution

As far as I investigated, the problem is that the composer max height is **immediately/instantly** calculated based on the existence of collections (such as images) in the app db (see `calc-max-height`). Whereas the composer is gradually animated.

The correct solution should probably be to only update the composer max height **after animations** have finished for all extra content.

@OmarBasem, I'm copying you in this thread just for awareness. I can try to fix this issue this week, but it's not gonna be a priority for me, so if you want/can, feel free to pick this up.

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/15981

### CODE REVIEW

**Review by ilmotta — APPROVED** [review]
The fix seems to work as far as I tested. Thanks @OmarBasem :+1: 

---

There are quite a few changes in the code, so could you consider adding more technical context in the PR's description next time? For instance, shortly explaining the solution, if there was any tradeoff, if you're satisfied and why, etc. Also the commit messages are just named `updates`, which hardly helps reviewers.

File: src/status_im2/contexts/chat/composer/effects.cljs [hunk 1]
```diff
@@ -186,6 +156,56 @@
        (rf/dispatch [:mention/on-change-text input-text])))
    [input-text]))
 
+(defn link-previews
```
Comment (flexsurfer) [inline src/status_im2/contexts/chat/composer/effects.cljs:159]: is this a new functionality? could you elaborate please

**Review by flexsurfer — APPROVED** [review]
its getting more and more complex, could you please tell in a few words what happened, and why so many complex changes in this PR, thanks

**Comment by OmarBasem** [comment]
@ilmotta @flexsurfer sorry for not giving an explanation.

We have 2 problems with the jumpy composer as shown in the videos in the [issue description](https://github.com/status-im/status-mobile/issues/15981#issue-1719870062). 

1. When removing links/images the top part of the composer jumps up.
2. When adding links/images the actions container at the bottom jumps up (can be seen in the second video at 0:05)

Fixing the second problem would fix both problems.

The second problem happens because when adding a link/image, there is an animation that happens. The links container starts at height of `0` then expands to `56`. The actions container jumps becuase when the height is zero it jumps up to fill the space. The solution for this problem is to move the links container and images container inside the input container instead of the whole composer container, and adjust the sheet container height to make space for links and images using effects.

File: src/status_im2/contexts/chat/composer/effects.cljs (hunk 1 above)
Comment (OmarBasem) [reply src/status_im2/contexts/chat/composer/effects.cljs:159]: Not new functionality. Just a different way of adjusting the links container height. Explained in comment below.

**Comment by VolodLytvynenko** [comment]
Hi @OmarBasem thank you for the fixes. No issues from my side. Can be merged

**Pull request merged**